### PR TITLE
Add Quantity flag helpers.

### PR DESCRIPTION
**Pull request by misterikkit**
**What type of PR is this?**

/kind feature
/sig cli

**What this PR does / why we need it**:
This revives the quantity flag helpers that were removed in https://github.com/kubernetes/kubernetes/pull/55511. I have only made minor tweaks to the original implementation.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/cli/flag/quantity_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func ExampleNewQuantityFlagValue() {
+	var myQuantity resource.Quantity
+	fs := pflag.NewFlagSet("test", pflag.ExitOnError)
+	fs.Var(NewQuantityFlagValue(&myQuantity), "amount", "how much is needed")
+	fs.Set("amount", "5Mi")
+	fmt.Println(myQuantity.Value())
+	// Output: 5242880
+}
```
Comment (stewart-yu) [inline staging/src/k8s.io/component-base/cli/flag/quantity_test.go]: thoes two of example left here seem stranger

**Comment by stewart-yu** [comment]
Does we need re-add?

File: staging/src/k8s.io/component-base/cli/flag/quantity_test.go (hunk 1 above)
Comment (misterikkit) [reply staging/src/k8s.io/component-base/cli/flag/quantity_test.go]: Could you explain what seems strange about them to you? The purpose of examples is to be instructional, so I want them to make sense to a wide audience.

**Comment by misterikkit** [comment]
> Does we really need re-add?

That's up to us/the SIGs. WG-Component-Standard is moving more things into "ComponentConfig" which is parsed from json or yaml rather than flag values. But there may still be times when a quantity flag is desirable. If we decide it doesn't belong here, I'm going to maintain a copy of this in private repos.

**Pull request closed without merging**